codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,18 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a {@code GOAWAY} frame was sent for the connection.
+         */
+        void onGoAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a {@code GOAWAY} was received from the remote endpoint. This
+         * event handler duplicates {@link Http2FrameListener#onGoAwayRead(io.netty.channel.ChannelHandlerContext,
+         * int, long, io.netty.buffer.ByteBuf)} but is added here in order to simplify application logic
+         * for handling {@code GOAWAY} in a uniform way. An application should generally not handle both events, but if
+         * it does this method is called first, before notifying the {@link Http2FrameListener}.
          */
-        void goingAway();
+        void onGoAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
Should we include the `lastStreamId` parameter in here, or is going to the `Endpoint` desired?